<!--REQUEST_ID=0e29bfe8-4c57-4f22-8c73-c350160395b5-->
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.47551&attempt_id=430ddb60-3700-4a41-9bd2-90c1640d98c1)

Attempted: snipes

<details><summary>Partial log (click to expand)</summary><p>

```
building
build flags: -j64 -l64 SHELL=/nix/store/fqm2x6kiay1q4vg7pqp4wp17bdijlyc3-bash-4.4-p23/bin/bash
g++ -o snipes -Werror -Wall -Wextra -std=c++11 -I/nix/store/iqa762mnxq71fj2lw5j98h8xdk1palfr-SDL2-2.0.8-dev/include/SDL2 -I/nix/store/iqa762mnxq71fj2lw5j98h8xdk1palfr-SDL2-2.0.8-dev/include/SDL2 -I/nix/store/08c0inrfxf8l6pf7mx5aqrwnkqi6m9xy-SDL2_ttf-2.0.14/include/SDL2 -I/nix/store/iqa762mnxq71fj2lw5j98h8xdk1palfr-SDL2-2.0.8-dev/include/SDL2 -I/nix/store/08c0inrfxf8l6pf7mx5aqrwnkqi6m9xy-SDL2_ttf-2.0.14/include/SDL2 -D_REENTRANT -O3 Snipes.cpp sdl/console.cpp sdl/keyboard.cpp sdl/sound.cpp sdl/timer.cpp -L/nix/store/pyvf271y923h7xjhymb1cpp24xzsjx16-SDL2-2.0.8/lib -Wl,-rpath,/nix/store/pyvf271y923h7xjhymb1cpp24xzsjx16-SDL2-2.0.8/lib -Wl,--enable-new-dtags -lSDL2 -lm -lSDL2_ttf
Snipes.cpp:2121:40: error: '__cdecl__' attribute directive ignored [-Werror=attributes]
 int __cdecl main(int argc, char* argv[])
                                        ^
cc1plus: all warnings being treated as errors
make: *** [GNUmakefile:10: snipes] Error 1
builder for '/nix/store/71g7hij1bjs66v9y08bm2v3ngsxyb61n-snipes-20171208.drv' failed with exit code 2
error: build of '/nix/store/71g7hij1bjs66v9y08bm2v3ngsxyb61n-snipes-20171208.drv' failed
```
</p></details>

--- a/pkgs/games/snipes/default.nix
+++ b/pkgs/games/snipes/default.nix
@@ -1,0 +1,45 @@
+{ stdenv, fetchFromGitHub, fetchurl, SDL2, SDL2_ttf }:
+
+let
+  font = fetchurl {
+    url    = "http://kingbird.myphotos.cc/ee22d44076adb8a34d8e20df4be3730a/SnipesConsole.ttf";
+    sha256 = "06n8gq18js0bv4svx84ljzhs9zmi81wy0zqcqj3b4g0rsrkr20a7";
+  };
+
+in stdenv.mkDerivation rec {
+  name = "snipes-${version}";
+  version = "20180930";
+
+  src = fetchFromGitHub {
+    owner  = "Davidebyzero";
+    repo   = "Snipes";
+    rev    = "343e14104b7848eb1f882401888e685b7918ef9f";
+    sha256 = "1rl70d5miak34warbwfv27z11vln4lvf7maqqc78z0gdc5zivdv2";
+  };
+
+  postPatch = ''
+    substitute config-sample.h config.h \
+      --replace SnipesConsole.ttf $out/share/snipes/SnipesConsole.ttf
+  '';
+
+  enableParallelBuilding = true;
+
+  buildInputs = [ SDL2 SDL2_ttf ];
+
+  installPhase = ''
+    runHook preInstall
+
+    install -Dm755 -t $out/bin snipes
+    install -Dm644 -t $out/share/doc/snipes *.md
+    install -Dm644 ${font} $out/share/snipes/SnipesConsole.ttf
+
+    runHook postInstall
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Modern port of the classic 1982 text-mode game Snipes";
+    homepage    = https://www.vogons.org/viewtopic.php?f=7&t=49073;
+    license     = licenses.free; # This reverse-engineered source code is released with the original authors' permission.
+    maintainers = with maintainers; [ peterhoeg ];
+  };
+}

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -1512,6 +1512,8 @@ with pkgs;
 
   simg2img = callPackage ../tools/filesystems/simg2img { };
 
+  snipes = callPackage ../games/snipes { };
+
   socklog = callPackage ../tools/system/socklog { };
 
   staccato = callPackage ../tools/text/staccato { };
@@ -1537,7 +1539,7 @@ with pkgs;
   riot-web = callPackage ../applications/networking/instant-messengers/riot/riot-web.nix {
     conf = config.riot-web.conf or null;
   };
-  
+
   roundcube = callPackage ../servers/roundcube { };
 
   rsbep = callPackage ../tools/backup/rsbep { };

